docs/source/onnx_dynamo.rst

```diff
@@ -0,0 +1,151 @@
+  - **ONNX Export Output**: :class:`ExportOutput` is the output of the exporter that contains the exported ONNX graph and diagnostics.
+  - **ONNX Export Output Serializer**: :class:`ExportOutputSerializer` serializes the exported model to a file.
+  - **ONNX Diagnostic Options**: :class:`DiagnosticOptions` has a set of options that control the diagnostics emitted by the exporter.
+
+Dependencies
+------------
+
+The ONNX exporter depends on extra Python packages not available through PyTorchs's `requirements.txt` file:
+
+  - `ONNX <https://onnx.ai>`_
+  - `ONNX Script <https://onnxscript.ai>`_
+
+They can be installed through `pip <https://pypi.org/project/pip/>`_ utility as follows:
+
+.. code-block:: bash
+
+  pip install onnx onnxscript-preview    # TODO: Replace with `onnxscript` when it is official
+
+A simple example
+----------------
+
+See below a demonstration of exporter API in action with a simple image classifier model as example:
+
+.. code-block:: python
+
+  import torch
+
+  class MNISTModel(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = torch.nn.Conv2d(1, 32, 3, 1, bias=False)
+        self.conv2 = torch.nn.Conv2d(32, 64, 3, 1, bias=False)
+        self.fc1 = torch.nn.Linear(9216, 128, bias=False)
+        self.fc2 = torch.nn.Linear(128, 10, bias=False)
+
+    def forward(self, tensor_x: torch.Tensor):
+        tensor_x = self.conv1(tensor_x)
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = self.conv2(tensor_x)
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = torch.nn.functional.max_pool2d(tensor_x, 2)
+        tensor_x = torch.flatten(tensor_x, 1)
+        tensor_x = self.fc1(tensor_x)
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = self.fc2(tensor_x)
+        output = torch.nn.functional.log_softmax(tensor_x, dim=1)
+        return output
+
+  tensor_x = torch.rand((64, 1, 28, 28), dtype=torch.float32)
+  export_output = torch.onnx.dynamo_export(MNISTModel(), tensor_x)
+
+As the code above shows, all you need is to provide ``torch.onnx.dynamo_export`` with an instance of the model and its input.
+The exporter will then return an instance of ``torch.onnx.ExportOutput`` that contains the exported ONNX graph along with extra information.
+
+The in-memory model (type: onnx.ModelProto) is available at ``export_output.model_proto`` following `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_,
+but the ONNX model can also be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the ``torch.onnx.ExportOutput.save`` API.
+
+.. code-block:: python
+
+  export_output.save("/path/to/mnist.onnx")
+
+Inspecting the ONNX model using GUI
+-----------------------------------
+
+You can view the exported model using `Netron <https://netron.app/>`__.
+
+.. image:: _static/img/onnx/onnx_dynamo_mnist_model.png
+    :width: 50%
+    :alt: MNIST model as viewed in Netron
```

Comment:
![image](https://github.com/pytorch/pytorch/assets/11205048/4d677796-d092-4281-a6d0-347464777bee)

This model may be a little too large for illustration. Should we remove some layers?

Reply:
will use MLP instead